cloud/scope/machine.go

```diff
@@ -148,12 +147,12 @@ func (m *MachineScope) Role() string {
 
 // GetInstanceID returns the GCPMachine instance id by parsing Spec.ProviderID.
 func (m *MachineScope) GetInstanceID() *string {
-	parsed, err := noderefutil.NewProviderID(m.GetProviderID()) //nolint: staticcheck
+	parsed, err := NewProviderID(m.GetProviderID()) //nolint: staticcheck
 	if err != nil {
 		return nil
 	}
 
-	return pointer.String(parsed.ID()) //nolint: staticcheck
+	return ptr.To[string](parsed.ID()) //nolint: staticcheck
```

Comment:
```suggestion
	return ptr.To(parsed.ID()) //nolint: staticcheck
```

All of these aren't needed, they're implicit

Reply:
Yeah you are right, although it's nice to have typing assertion there. I think it might be worth keeping them